detector/interface.go

```diff
@@ -27,6 +27,18 @@ type MasterChanged interface {
 	OnMasterChanged(*mesos.MasterInfo)
 }
 
+// optional interface that, if implemented by the same object as implements
+// MasterChanged, will receive an additional callbacks independently of
+// leadership changes. it's possible that, as a result of a leadership change,
+// both the OnMasterChanged and UpdatedMasters callbacks would be invoked.
+// the additional callback, UpdatedMasters, is useful to clients that wish to know
+// the entire set of Mesos masters currently running.
+// **NOTE:** Detector implementations are not required to support this optional
+// interface. Please RTFM of the detector implementation that you want to use.
+type AllMasters interface {
+	UpdatedMasters([]*mesos.MasterInfo)
```

Comment:
It would, except they would miss events if they were too slow. Yes we could
use a buffered chan .. how big is big enough if a client has disappeared?

On Sat, May 30, 2015 at 4:09 AM, Tomás Senart <notifications@github.com>
wrote:

> In detector/interface.go
> <https://github.com/mesos/mesos-go/pull/111#discussion_r31378872>:
>
> > @@ -27,6 +27,18 @@ type MasterChanged interface {
> >  	OnMasterChanged(*mesos.MasterInfo)
> >  }
> >
> > +// optional interface that, if implemented by the same object as implements
> > +// MasterChanged, will receive an additional callbacks independently of
> > +// leadership changes. it's possible that, as a result of a leadership change,
> > +// both the OnMasterChanged and UpdatedMasters callbacks would be invoked.
> > +// the additional callback, UpdatedMasters, is useful to clients that wish to know
> > +// the entire set of Mesos masters currently running.
> > +// **NOTE:** Detector implementations are not required to support this optional
> > +// interface. Please RTFM of the detector implementation that you want to use.
> > +type AllMasters interface {
> > +	UpdatedMasters([]*mesos.MasterInfo)
>
> Wouldn't a non-blocking send solve the problem of readers disappearing or
> being slow?
>
> select {case eventch <- ev:default:
>     // reader is too slow, drop event
> }
>
> If we want to support some lag from readers, we can tune the channel
> buffer to some value bigger than zero.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/111/files#r31378872>.
>